packages/db/tests/async-cleanup-race.test.ts

```diff
@@ -0,0 +1,111 @@
+import { describe, expect, it, vi } from "vitest"
+import { createCollection } from "../src/collection.js"
+
+type Row = { id: string; name: string }
+
+// Simulate an external observer keyed by collection id (like a query observer)
+// The cleanup() will cancel whatever observer is currently active for that id
+const observers = new Map<
+  string,
+  {
+    fetchTimer: ReturnType<typeof setTimeout> | null
+    cancelled: { value: boolean }
+  }
+>()
+const startCounts = new Map<string, number>()
+
+describe(`Async sync cleanup race with GC`, () => {
+  it(`reproduces: GC cleanup cancels the restarted sync, leaving collection stuck empty`, async () => {
+    vi.useFakeTimers()
+
+    try {
+      const collection = createCollection<Row>({
```

Comment:
The definition of the collection is identical to the one in `async-cleanup-livequery-race.test.ts`. Extract it to a constant/function in a helper file.